### PR TITLE
[sharedb] Make version argument nullable

### PR DESCRIPTION
**Pull request by ElArabeLoco**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: (https://share.github.io/sharedb/api/connection#fetchsnapshot)

### CODE REVIEW

File: types/sharedb/lib/client.d.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ export class Connection extends ShareDB.TypedEmitter<ShareDB.ConnectionEventMap>
     get(collectionName: string, documentID: string): Doc;
     createFetchQuery<T = any>(collectionName: string, query: any, options?: {results?: Array<Doc<T>>} | null, callback?: (err: Error, results: Array<Doc<T>>) => void): Query<T>;
     createSubscribeQuery<T = any>(collectionName: string, query: any, options?: {results?: Array<Doc<T>>} | null, callback?: (err: Error, results: Array<Doc<T>>) => void): Query<T>;
-    fetchSnapshot(collection: string, id: string, version: number, callback: (error: Error, snapshot: ShareDB.Snapshot) => void): void;
+    fetchSnapshot(collection: string, id: string, version: number | null, callback: (error: Error, snapshot: ShareDB.Snapshot) => void): void;
```
Comment (alecgibson) [inline types/sharedb/lib/client.d.ts:29]: While we're here, can we please also make `fetchSnapshotByTimestamp()`'s `timestamp` argument [nullable](https://github.com/share/sharedb/blob/082e27b973531845bc42f9a8f27401eee063e0fc/test/client/snapshot-timestamp-request.js#L494-L495)?

**Review by alecgibson — CHANGES_REQUESTED** [review]

**Review by alecgibson — APPROVED** [review]
Thanks for this!

File: types/sharedb/lib/client.d.ts (hunk 1 above)
Comment (ElArabeLoco) [reply types/sharedb/lib/client.d.ts:29]: Sure! Thanks for reviewing.

**Comment by ElArabeLoco** [comment]
Ready to merge

**Pull request merged**